c2corg_ui/static/js/imageuploader.js

```diff
@@ -278,31 +286,22 @@ app.ImageUploaderController.prototype.save = function() {
   var defer = this.q_.defer();
 
   this.api_.createImages(this.files, this.documentService.document)
-  .then(function() {
+  .then(function(data) {
     var meta;
     var id;
+    var images = data['config']['data']['images'];
 
     $('.img-container').each(function(i, image) {
-      meta = this.files[i]['metadata'];
+      meta = images[i]['metadata'];
```

Comment:
> Perhaps the correct code would rather be

I have pushed a fixup commit about this issue, see 4cb360c
It seems to work but it would be cool if someone could doublecheck.